internal/core/domain/merkleTreeNode.go

```diff
@@ -0,0 +1,9 @@
+package domain
+
+type MerkleTreeNode struct {
+	MTID   int64
```

Comment:
I guess you don't need this type. Because you use only MTID field.

Reply:
By the way, there is this one: https://github.com/iden3/go-merkletree-sql/blob/master/node.go#L27
But it doesn't have mt id field.